backend/Squirrel.Core/Squirrel.Core.BLL/Services/AuthService.cs

```diff
@@ -29,10 +32,24 @@ public async Task<RefreshedAccessTokenDto> LoginAsync(UserLoginDto userLoginDto)
 
     public async Task<RefreshedAccessTokenDto> RegisterAsync(UserRegisterDto userRegisterDto)
     {
-        // Dummy user info.
-        var userId = 0;
+        if (await _context.Users.FirstOrDefaultAsync(u => u.Username == userRegisterDto.Username) is not null)
+        {
+            throw new UsernameAlreadyRegisteredException();
+        }
+        
+        if (await _context.Users.FirstOrDefaultAsync(u => u.Email == userRegisterDto.Email) is not null)
+        {
+            throw new EmailAlreadyRegisteredException();
+        }
+
+        var newUser = _mapper.Map<User>(userRegisterDto)!;
+        var salt = BytesGenerator.GetRandomBytes();
+        newUser.Salt = Convert.ToBase64String(salt);
```

Comment:
Can you create function in passwordProcessor (I don't really like naming) to generate salt in byte64 format?